test/new-e2e/tests/npm/eks_1host_test.go

```diff
@@ -0,0 +1,130 @@
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/components"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments"
+	enveks "github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments/aws/eks"
+	"github.com/DataDog/test-infra-definitions/common/config"
+	npmtools "github.com/DataDog/test-infra-definitions/components/datadog/apps/npm-tools"
+	"github.com/DataDog/test-infra-definitions/components/datadog/kubernetesagentparams"
+	"github.com/DataDog/test-infra-definitions/components/docker"
+	kubeComp "github.com/DataDog/test-infra-definitions/components/kubernetes"
+	"github.com/DataDog/test-infra-definitions/resources/aws"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/ec2"
+
+	"github.com/pulumi/pulumi-kubernetes/sdk/v4/go/kubernetes"
+)
+
+type eksHttpbinEnv struct {
+	environments.EKS
+
+	// Extra Components
+	HTTPBinHost *components.RemoteHost
+}
+
+type eksVMSuite struct {
+	e2e.BaseSuite[eksHttpbinEnv]
+}
+
+func eksHttpbinEnvProvisioner() e2e.PulumiEnvRunFunc[eksHttpbinEnv] {
+	return func(ctx *pulumi.Context, env *eksHttpbinEnv) error {
+		awsEnv, err := aws.NewEnvironment(ctx)
+		if err != nil {
+			return err
+		}
+		env.EKS.AwsEnvironment = &awsEnv
+
+		vmName := "httpbinvm"
+		httpbinHost, err := ec2.NewVM(awsEnv, vmName)
+		if err != nil {
+			return err
+		}
+		err = httpbinHost.Export(ctx, &env.HTTPBinHost.HostOutput)
+		if err != nil {
+			return err
+		}
+
+		// install docker.io
+		manager, _, err := docker.NewManager(*awsEnv.CommonEnvironment, httpbinHost, true)
+		if err != nil {
+			return err
+		}
+
+		composeContents := []docker.ComposeInlineManifest{dockerHTTPBinCompose()}
+		_, err = manager.ComposeStrUp("httpbin", composeContents, pulumi.StringMap{})
+		if err != nil {
+			return err
+		}
+
+		npmToolsWorkload := func(e config.CommonEnvironment, kubeProvider *kubernetes.Provider) (*kubeComp.Workload, error) {
+			// NPM tools Workload
+			testURL := "http://" + env.HTTPBinHost.Address + "/"
+			return npmtools.K8sAppDefinition(*awsEnv.CommonEnvironment, kubeProvider, "npmtools", testURL)
+		}
+
+		params := enveks.GetProvisionerParams(
+			enveks.WithEKSLinuxNodeGroup(),
+			enveks.WithAgentOptions(kubernetesagentparams.WithHelmValues(systemProbeConfigNPMHelmValues)),
+			enveks.WithWorkloadApp(npmToolsWorkload),
+		)
+		enveks.Run(ctx, &env.EKS, params)
+
+		return nil
+	}
+}
+
+// TestEKSVMSuite will validate running the agent
+func TestEKSVMSuite(t *testing.T) {
+	s := &eksVMSuite{}
+	e2eParams := []e2e.SuiteOption{e2e.WithProvisioner(e2e.NewTypedPulumiProvisioner("eksHttpbin", eksHttpbinEnvProvisioner(), nil))}
+
+	e2e.Run(t, s, e2eParams...)
+}
+
+// BeforeTest will be called before each test
+func (v *eksVMSuite) BeforeTest(suiteName, testName string) {
+	v.BaseSuite.BeforeTest(suiteName, testName)
+
+	// default is to reset the current state of the fakeintake aggregators
+	if !v.BaseSuite.IsDevMode() {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Does this mean you don't need to clean up the fakeintake if you are in DevMode?

Reply:
no It's just for testing purpose, doesn't need to wait on all tests every time, and I can check the fakeintake server afterwards for debugging